api/models/Subscribers.js

```diff
@@ -178,6 +178,16 @@ subscribersSchema.path('transaction').validate(async function(transaction) {
   return true;
 }, 'transaction puchase token error');
 
+subscribersSchema.post('findOneAndUpdate', async function(subscriber) {
+  const status = subscriber?.transaction?.subscriptionState || 'not_subscribed';
+  try {
+    const doc = await subscriber.model("Subscribers",subscribersSchema).findById(subscriber._id)
+    await doc.update({status});
```

Comment:
DeprecationWarning: collection.update is deprecated. Use updateOne, updateMany, or bulkWrite instead.